lightning/src/ln/channelmanager.rs

```diff
@@ -656,25 +627,74 @@ pub struct ChannelDetails {
+	///
+	/// [`inbound_capacity_msat`]: ChannelDetails::inbound_capacity_msat
+	pub to_remote_reserve_satoshis: u64,
 	/// The user_id passed in to create_channel, or 0 if the channel was inbound.
 	pub user_id: u64,
 	/// The available outbound capacity for sending HTLCs to the remote peer. This does not include
 	/// any pending HTLCs which are not yet fully resolved (and, thus, who's balance is not
 	/// available for inclusion in new outbound HTLCs). This further does not include any pending
 	/// outgoing HTLCs which are awaiting some other resolution to be sent.
+	///
+	/// This value is not exact. Due to various in-flight changes, feerate changes, and our
+	/// conflict-avoidance policy, exactly this amount is not likely to be spendable. However, we
+	/// should be able to spend nearly this amount.
 	pub outbound_capacity_msat: u64,
 	/// The available inbound capacity for the remote peer to send HTLCs to us. This does not
 	/// include any pending HTLCs which are not yet fully resolved (and, thus, who's balance is not
 	/// available for inclusion in new inbound HTLCs).
 	/// Note that there are some corner cases not fully handled here, so the actual available
 	/// inbound capacity may be slightly higher than this.
+	///
+	/// This value is not exact. Due to various in-flight changes, feerate changes, and our
+	/// counterparty's conflict-avoidance policy, exactly this amount is not likely to be spendable.
+	/// However, our counterparty should be able to spend nearly this amount.
 	pub inbound_capacity_msat: u64,
+	/// The number of required confirmations on the funding transaction before the funding will be
+	/// considered "locked". This number is selected by the channel fundee (i.e. us if
+	/// [`is_outbound`] is *not* set), and can be selected for inbound channels with
+	/// [`ChannelHandshakeConfig::minimum_depth`] or limited for outbound channels with
+	/// [`ChannelHandshakeLimits::max_minimum_depth`].
+	///
+	/// This value will be `None` for outbound channels until the counterparty accepts the channel.
+	///
+	/// [`is_outbound`]: ChannelDetails::is_outbound
+	/// [`ChannelHandshakeConfig::minimum_depth`]: crate::util::config::ChannelHandshakeConfig::minimum_depth
+	/// [`ChannelHandshakeLimits::max_minimum_depth`]: crate::util::config::ChannelHandshakeLimits::max_minimum_depth
+	pub confirmations_required: Option<u32>,
+	/// The number of blocks (after our commitment transaction confirms) that we will need to wait
+	/// until we can claim our funds after we force-close the channel. During this time our
+	/// counterparty is allowed to punish us if we broadcasted a stale state. If our counterparty
+	/// force-closes the channel and broadcasts a commitment transaction we do not have to wait any
+	/// time to claim our non-HTLC-encumbered funds.
+	///
+	/// This value will be `None` for outbound channels until the counterparty accepts the channel.
+	pub to_self_delay: Option<u16>,
```

Comment:
What do you think of `on_holder_tx_csv` ?

Reply:
I admit I kinda like using the same name as the spec, but alternatively we could just make it more verbose and go with `spend_csv_on_our_broadcasted_commitment` or so?

Reply:
Yeap, better to be verbose here than avoid accidental inverting of values not catched up during review :/ Though yeah we can address this latter.

Reply:
Its a public interface, lets get it right :)

Reply:
How about `force_close_spend_delay`?

That way the naming is in terms of preexisting interface terminology rather than esoteric protocol details.